### PR TITLE
TER-89 Added Terraform Platform example with ECS, Postgres and Redis

### PR DESCRIPTION
**Pull request by shwetanshu-cldcvr**

### CODE REVIEW

File: examples/platform-demo/platform/large.tfvars
Comment (petmal) [inline examples/platform-demo/platform/large.tfvars:20]: Just wondering. Should this really be in the profile? I guess it depends on how we define the profile (i.e. by performance - small/large or by kind dev/prod). But what if the user needs large-dev etc. It quickly becomes multi-dimensional.
Unless we support multiple profiles that complement each other (e.g. generate --profile def --profile large). The assumption is that the profiles would not overlap or they would end up overriding each other.

Comment (petmal) [reply examples/platform-demo/platform/large.tfvars:20]: @kanaksinghal @dshelley66

**Comment by petmal** [comment]
It looks good to me. It certainly was lot of work going into it. In this case we start from "not-so-useful" Terraform registry modules. Hopefully better designed starting modules would reduce the amount of work needed.

File: examples/platform-demo/modules/alb/alb_main.tf
Comment (petmal) [inline examples/platform-demo/modules/alb/alb_main.tf:91]: So there is a bucket for each elb. Just wondering would people generally create new buckets or reuse them between resources - say for logging.
I guess at some point we may need a storage component that would be a reusable-user-facing bucket implementation.

File: examples/platform-demo/readme.md
Comment (kanaksinghal) [inline examples/platform-demo/readme.md:4]:
```suggestion
rm -rf .terrarium && terrarium generate -p ./platform -a ./banking -o ./.terrarium
```

File: .gitignore
Comment (kanaksinghal) [inline .gitignore:48]: it's already there on line 36

```suggestion
```

**Comment by dshelley66** [comment]
@shwetanshu-cldcvr @kanaksinghal (This comment isn't intended to block the merge of this work). I think it would be useful to document what steps went into making this. I can see how to generate the outcome (i.e. a working TF template) - I did this and it works. What I don't know is:
1. Which of the files in this PR were "manually" written and which generated by terrarium tooling?
2. I see small and large tfvars but have no idea how they are used
3. I'm not clear on if there are variables I'm can (or required to) provide on generate

Maybe the answers to these questions exist somewhere and I just don't know where. Basically, if I want to make my own platform - how to start etc...

thx

**Comment by kanaksinghal** [comment]
> 1. Which of the files in this PR were "manually" written and which generated by terrarium tooling?

I believe Shwetanshu didn't use the terrarium vs-code extension for this yet. It wasn't ready for use at the time.

> 2. I see small and large tfvars but have no idea how they are used

This will be used after [TER-230](https://cldcvr.atlassian.net/browse/TER-230) & [TER-251](https://cldcvr.atlassian.net/browse/TER-251)

> 3. I'm not clear on if there are variables I'm can (or required to) provide on generate

no, if you see `terrarium generate --help` it doesn't have any variables option. All configuration comes from the app yaml. You can refer variables.tf file for terraform variables been declared and you need to resolve at the time of running the terraform apply command

File: examples/platform-demo/platform/large.tfvars
Comment (kanaksinghal) [reply examples/platform-demo/platform/large.tfvars:20]: you are right, it might make more sense to have dev/staging/prod profiles. for now (with one profile model), it's still possible to create a profile say large-dev. we can consider adding multi dimensional profiles in future as we encounter the use-case.

**Review by kanaksinghal — APPROVED** [review]

File: examples/platform-demo/modules/alb/alb_main.tf
Comment (rajatsharma111) [inline examples/platform-demo/modules/alb/alb_main.tf:2]: @shwetanshu-cldcvr should we keep these modules in farm?

File: examples/platform-demo/platform/large.tfvars
Comment (rajatsharma111) [inline examples/platform-demo/platform/large.tfvars:34]: @shwetanshu-cldcvr should we replace these names to a generic one?

File: examples/platform-demo/platform/small.tfvars
Comment (rajatsharma111) [inline examples/platform-demo/platform/small.tfvars:34]: @shwetanshu-cldcvr should we replace these names to a generic one?

File: examples/platform-demo/platform/variables.tf
Comment (rajatsharma111) [inline examples/platform-demo/platform/variables.tf:27]: @shwetanshu-cldcvr should we replace these names to a generic one?

File: examples/platform-demo/platform/variables.tf
Comment (rajatsharma111) [inline examples/platform-demo/platform/variables.tf:78]: @shwetanshu-cldcvr should we replace these names to a generic one?

File: examples/platform-demo/platform/small.tfvars
Comment (rajatsharma111) [inline examples/platform-demo/platform/small.tfvars:9]: @shwetanshu-cldcvr should we call it as testing or staging instead of destroy?

**Comment by rajatsharma111** [comment]
@shwetanshu-cldcvr can we merge this PR if all good?

**Comment by dshelley66** [comment]
Is the intention to get this merged soon? @shwetanshu-cldcvr is on leave for 2 weeks after tomorrow.....

**Review by rajatsharma111 — DISMISSED** [review]

**Review by dshelley66 — DISMISSED** [review]

**Review by rajatsharma111 — APPROVED** [review]

**Pull request merged**
